src/commands/logstream/startStreamingLogs.ts

```diff
@@ -38,12 +38,12 @@ export async function startStreamingLogs(context: vscode.ExtensionContext, tree:
             await treeItem.client.updateLogsConfig(logsConfig);
         }
 
-        if (!treeItem.logStreamOutputChannel) {
+        if (logsOutputChannels[treeItem.logStreamLabel] == null) {
```

Comment:
Wouldn't this be undefined instead of null? In any case, I prefer `!logsOutputChannels[treeItem.logStreamLabel]` to cover all bases

Reply:
Also you should change the type of `channelObj` to `vscode.OutputChannel | undefined`.

The linter doesn't like `!<vscode.OutputChannel>` since theoretically that would always be defined/true. However, it makes sense to have `!<vscode.OutputChannel | undefined>` because it could be true or false

Reply:
Although counter intuitive, actually, `==null` covers both cases (still, I changed it to: `!logsOutputChannels[treeItem.logStreamLabel]`):

_Debug console output:_
`logsOutputChannels[treeItem.logStreamLabel]
undefined
(logsOutputChannels[treeItem.logStreamLabel] == null)
true
(logsOutputChannels[treeItem.logStreamLabel] == undefined)
true`

Also, did not use `vscode.OutputChannel | undefined` because `appservice.startStreamingLogs` does not accept it:
`export declare function startStreamingLogs(client: SiteClient, outputChannel: vscode.OutputChannel, path?: string): Promise<ILogStream>;`

Let me know if you have another idea, I am open :)

Reply:
Changed back to `logsOutputChannels[treeItem.logStreamLabel] == null` due to the reason you mentioned.

Reply:
That's an interesting point about `== null`. However, we prefer `===` instead of `==`, which would have different behavior. We even have a "triple-equals" tslint rule specified [here](https://github.com/Microsoft/vscode-azurefunctions/blob/master/tslint.json#L99), but I guess we currently have `allow-null-check` specified. I think it's weird to allow that for just one counterintuitive edge case, so we will probably disallow it going forward.

Here's what I suggest to make `vscode.OutputChannel | undefined` work:
```typescript
let outputChannel: vscode.OutputChannel | undefined = logsOutputChannels[treeItem.logStreamLabel].channelObj;
if (!outputChannel) {
    outputChannel = vscode.window.createOutputChannel(localize('logStreamLabel', '{0} - Log Stream', treeItem.logStreamLabel));
    context.subscriptions.push(outputChannel);
    logsOutputChannels[treeItem.logStreamLabel] = { channelObj: outputChannel };
}
treeItem.logStream = await appservice.startStreamingLogs(treeItem.client, outputChannel, treeItem.logStreamPath);
```

Reply:
Makes total sense, thanks!